### DISCUSSION TITLE
Quais são as (des)vantagens de se ter uma aplicação sem backend?

### DISCUSSION THREAD

**Discussion by mumumilk** [frontendbr/forum#874] [Perguntas e Respostas]
Na opnião de vocês, quais são as vantagens e desvantagens de se trabalhar numa aplicação assim?

Levando em conta que hoje temos funções e banco/armazenamento em nuvem, até que grau de complexidade de um sistema seria suportável fazê-lo sem ajuda do backend?

**Comment by klarkc** [comment]
Essa é minha opinião, já estudei muito arquitetura de software serverless, porém até agora tive poucas oportunidades reais de trabalhar com ela. Mas acho que as soluções já estão maduras o suficiente, e bem alinhadas com a proposta das Progressive Web Apps.

### Vantagens

- Não se preocupar com infra estrutura e escalonamento.
- No desenvolvimento do software, você deixa as complexidades de lidar com decisões de arquitetura mais pra frente e se preocupa mais com a funcionalidade, calha bem com um modelo MVP.

### Desvantagens

- Algumas vezes você (pode ou não) precisar integrar serviços muito diferentes na sua aplicação, isso pode ser custoso financeiramente (afinal o tempo estudando a integração desses serviços é dinheiro), mas eu diria que isso não é uma desvantagem, numa arquitetura orientada a micro-serviços, isso na verdade é uma vantagem.
- Após um tempo, a sua aplicação fica mais pesada no front, portanto é essencial, em algum momento, se preocupar com a performance dela, e isso muitas vezes exige boas práticas de programação e conhecimentos profundos de arquitetura de software e novas tecnologias criadas para resolverem esses problemas, como o protocolo HTTP2, lazy loading, etc.

**Comment by felipefialho** [comment]
[Off] @klarkc Estamos usando Serverless aqui no Cubo, inclusive [montamos um Meetup sobre o assunto](https://www.meetup.com/Serverless-SP/), olha lá depois 😄
